diff --git a/hardware/libhardware_legacy/wifi/Android.mk b/hardware/libhardware_legacy/wifi/Android.mk
index c7bdc59..53b9e44 100644
--- a/hardware/libhardware_legacy/wifi/Android.mk
+++ b/hardware/libhardware_legacy/wifi/Android.mk
@@ -45,5 +45,10 @@ LOCAL_SRC_FILES += wifi/wifi.c
 ifeq ($(BOARD_HAVE_SAMSUNG_WIFI),true)
 LOCAL_CFLAGS += -DSAMSUNG_WIFI
 endif
+ifeq ($(WIFI_MAC_ADDR_CSPSA),true)
+LOCAL_CFLAGS += -DCSPSA
+LOCAL_C_INCLUDES += hardware/snda/modem/cspsa
+LOCAL_SHARED_LIBRARIES += libcspsa
+endif
 
 LOCAL_SHARED_LIBRARIES += libnetutils
diff --git a/hardware/libhardware_legacy/wifi/wifi.c b/hardware/libhardware_legacy/wifi/wifi.c
index 5c8a11f..cdcc6c3 100644
--- a/hardware/libhardware_legacy/wifi/wifi.c
+++ b/hardware/libhardware_legacy/wifi/wifi.c
@@ -136,6 +136,88 @@ static const char MODULE_FILE[]         = "/proc/modules";
 static const char IFNAME[]              = "IFNAME=";
 #define IFNAMELEN			(sizeof(IFNAME) - 1)
 static const char WPA_EVENT_IGNORE[]    = "CTRL-EVENT-IGNORE ";
+#ifdef CSPSA
+#include "cspsa.h"
+
+static int wifi_fetch_mac(char* wifi_driver_param)
+{
+    int found = 0;
+    int ret = -1;
+    CSPSA_Result_t result;
+    CSPSA_Handle_t handle;
+    CSPSA_Size_t size;
+    CSPSA_Key_t key = 0x00010000;
+    CSPSA_Data_t *cspsa_data = NULL;
+
+    result = CSPSA_Open("CSPSA0", &handle);
+    if (result != T_CSPSA_RESULT_OK) {
+        ALOGE("Can't open CSPSA area (result 0x%X) ", result);
+        goto cspsa_finished;
+    }
+
+    result = CSPSA_GetSizeOfValue(handle, key, &size);
+    if (result != T_CSPSA_RESULT_OK) {
+        ALOGE("Can't get size of key (h %p key 0x%x result 0x%X).",
+          handle, key, result);
+        goto cspsa_finished;
+    }
+
+    if (size != 6) {
+        ALOGE("Read wrong amount of bytes: %d", size);
+        goto cspsa_finished;
+    }
+
+    cspsa_data = (CSPSA_Data_t *) malloc(size);
+    if (!cspsa_data) {
+        ALOGE("Can't malloc %d bytes.", size);
+        goto cspsa_finished;
+    }
+
+    result = CSPSA_ReadValue(handle, key, size, cspsa_data);
+    if (result != T_CSPSA_RESULT_OK) {
+        ALOGE("Can't read from CSPSA (h %p  key 0x%x size %d res 0x%X).",
+            handle, key, size, result);
+        goto cspsa_finished;
+    }
+
+    /* Don't use CSPSA MAC address if it ends with :00:00:00.*/
+    if ((0 == cspsa_data[3]) && (0 == cspsa_data[4]) && (0 == cspsa_data[5])) {
+        ALOGI("Default MAC address read from CSPSA. Using random.");
+        goto cspsa_finished;
+    }
+
+    found = 1;
+    ret = 0; //success
+    memcpy(wifi_driver_param, cspsa_data, 6);
+    ALOGI("MAC address from CSPSA [0x%02x:0x%02x:0x%02x:0x%02x:0x%02x:0x%02x]",
+         cspsa_data[0], cspsa_data[1], cspsa_data[2],
+         cspsa_data[3], cspsa_data[4], cspsa_data[5]);
+
+cspsa_finished:
+    if (cspsa_data)
+        free(cspsa_data);
+    result = CSPSA_Close(&handle);
+    if (result != T_CSPSA_RESULT_OK)
+        ALOGE("Can't close CSPSA area (result %x) ", result);
+
+    if (found == 0) {
+      memcpy(wifi_driver_param, "macaddr",
+              sizeof(0x00));
+    }
+    return ret;
+}
+
+static void wifi_format_mac_param(char* mac)
+{
+    char mac_address[100];
+    sprintf(mac_address, "0x%02x,0x%02x,0x%02x,0x%02x,0x%02x,0x%02x",
+        mac[0], mac[1], mac[2], mac[3], mac[4], mac[5]);
+    mac[0] = ' '; //one space to separate insmod parameters
+    strcpy(&mac[1], "macaddr");
+    strcat(mac, "=");
+    strcat(mac, mac_address);
+}
+#endif
 
 static const char SUPP_ENTROPY_FILE[]   = WIFI_ENTROPY_FILE;
 static unsigned char dummy_key[21] = { 0x02, 0x11, 0xbe, 0x33, 0x43, 0x35,
@@ -187,8 +269,21 @@ static int insmod(const char *filename, const char *args)
     if (!module)
         return -1;
 
+#ifndef CSPSA
     ret = init_module(module, size, args);
+#else
+    char driver_module_arg[255];
+    char mac_address_param[255];
+    strcpy(driver_module_arg, args);
 
+    if (0 == wifi_fetch_mac(mac_address_param)) {
+        wifi_format_mac_param(mac_address_param);
+    }
+    strcat(driver_module_arg, mac_address_param);
+    ALOGI("***wlan.ko (%s)", driver_module_arg);
+
+    ret = init_module(module, size, driver_module_arg);
+#endif
     free(module);
 
     return ret;
