diff --git a/hardware/libhardware_legacy/include/hardware_legacy/AudioSystemLegacy.h b/hardware/libhardware_legacy/include/hardware_legacy/AudioSystemLegacy.h
index b0ca0b3..a3f7746 100644
--- a/hardware/libhardware_legacy/include/hardware_legacy/AudioSystemLegacy.h
+++ b/hardware/libhardware_legacy/include/hardware_legacy/AudioSystemLegacy.h
@@ -65,7 +65,7 @@ enum audio_source {
     AUDIO_SOURCE_CAMCORDER = 5,
     AUDIO_SOURCE_VOICE_RECOGNITION = 6,
     AUDIO_SOURCE_VOICE_COMMUNICATION = 7,
-#ifdef QCOM_HARDWARE
+#if defined(QCOM_HARDWARE) || defined(STE_FM)
     AUDIO_SOURCE_REMOTE_SUBMIX = 8,
     AUDIO_SOURCE_FM_RX = 9,
     AUDIO_SOURCE_FM_RX_A2DP = 10,
@@ -92,8 +92,9 @@ public:
         ENFORCED_AUDIBLE = 7, // Sounds that cannot be muted by user and must be routed to speaker
         DTMF             = 8,
         TTS              = 9,
-#ifdef QCOM_HARDWARE
+#if defined(QCOM_HARDWARE) || defined(STE_FM)
         INCALL_MUSIC     = 10,
+	FM               = 11,
 #endif
         NUM_STREAM_TYPES
     };
@@ -257,7 +258,7 @@ public:
         DEVICE_OUT_AUX_DIGITAL = 0x400,
         DEVICE_OUT_ANLG_DOCK_HEADSET = 0x800,
         DEVICE_OUT_DGTL_DOCK_HEADSET = 0x1000,
-#ifdef QCOM_HARDWARE
+#if defined(QCOM_HARDWARE) || defined(STE_FM)
         DEVICE_OUT_USB_ACCESSORY = 0x2000,
         DEVICE_OUT_USB_DEVICE = 0x4000,
         DEVICE_OUT_FM = 0x8000,
@@ -274,7 +275,7 @@ public:
                 DEVICE_OUT_BLUETOOTH_SCO_CARKIT | DEVICE_OUT_BLUETOOTH_A2DP | DEVICE_OUT_BLUETOOTH_A2DP_HEADPHONES |
                 DEVICE_OUT_BLUETOOTH_A2DP_SPEAKER | DEVICE_OUT_AUX_DIGITAL |
                 DEVICE_OUT_ANLG_DOCK_HEADSET | DEVICE_OUT_DGTL_DOCK_HEADSET |
-#ifdef QCOM_HARDWARE
+#if defined(QCOM_HARDWARE) || defined(STE_FM)
                 DEVICE_OUT_USB_ACCESSORY | DEVICE_OUT_USB_DEVICE |
                 DEVICE_OUT_ANC_HEADSET | DEVICE_OUT_ANC_HEADPHONE |
                 DEVICE_OUT_FM | DEVICE_OUT_FM_TX |
@@ -284,12 +285,12 @@ public:
 #endif
         DEVICE_OUT_ALL_A2DP = (DEVICE_OUT_BLUETOOTH_A2DP | DEVICE_OUT_BLUETOOTH_A2DP_HEADPHONES |
                 DEVICE_OUT_BLUETOOTH_A2DP_SPEAKER),
-#ifdef QCOM_HARDWARE
+#if defined(QCOM_HARDWARE) || defined(STE_FM)
         DEVICE_OUT_ALL_USB = (DEVICE_OUT_USB_ACCESSORY | DEVICE_OUT_USB_DEVICE),
 #endif
 
         // input devices
-#ifdef QCOM_HARDWARE
+#if defined(QCOM_HARDWARE) || defined(STE_FM)
         DEVICE_IN_COMMUNICATION = 0x100000,
         DEVICE_IN_AMBIENT = 0x200000,
         DEVICE_IN_BUILTIN_MIC = 0x400000,
@@ -299,7 +300,11 @@ public:
         DEVICE_IN_VOICE_CALL = 0x4000000,
         DEVICE_IN_BACK_MIC = 0x8000000,
         DEVICE_IN_ANC_HEADSET = 0x10000000,
+#ifdef STE_FM
+	DEVICE_IN_FM_RX = 0x1000000,
+#else
         DEVICE_IN_FM_RX = 0x20000000,
+#endif
         DEVICE_IN_FM_RX_A2DP = 0x40000000,
         DEVICE_IN_DEFAULT = DEVICE_IN_BUILTIN_MIC,
         DEVICE_IN_ANLG_DOCK_HEADSET = 0x80000000,
@@ -318,7 +323,7 @@ public:
 
         DEVICE_IN_ALL = (DEVICE_IN_COMMUNICATION | DEVICE_IN_AMBIENT | DEVICE_IN_BUILTIN_MIC |
                 DEVICE_IN_BLUETOOTH_SCO_HEADSET | DEVICE_IN_WIRED_HEADSET | DEVICE_IN_AUX_DIGITAL |
-#ifdef QCOM_HARDWARE
+#if defined(QCOM_HARDWARE) || defined(STE_FM)
                 DEVICE_IN_VOICE_CALL | DEVICE_IN_BACK_MIC | DEVICE_IN_ANC_HEADSET |
                 DEVICE_IN_FM_RX | DEVICE_IN_FM_RX_A2DP | DEVICE_IN_DEFAULT |
                 DEVICE_IN_ANLG_DOCK_HEADSET | DEVICE_IN_PROXY)
